server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/HealthCheckRoutes.java

```diff
@@ -91,25 +123,25 @@ public void define(Service service) {
 
     public Object validateHealthChecks(Request request, Response response) {
         Set<ComponentName> selectedComponentNames = getComponentNames(request);
-        boolean isStrict = isStrictMode(request);
+        StatusCodeConvertMode convertMode = getStatusCodeConvertMode(request);
         Collection<HealthCheck> selectedHealthChecks = selectHealthChecks(selectedComponentNames);
         List<Result> results = executeHealthChecks(selectedHealthChecks).collectList().block();
         ResultStatus status = retrieveAggregationStatus(results);
-        response.status(getCorrespondingStatusCode(status, isStrict));
+        response.status(getCorrespondingStatusCode(status, convertMode));
         return new HeathCheckAggregationExecutionResultDto(status, mapResultToDto(results));
     }
 
     public Object performHealthCheckForComponent(Request request, Response response) {
         String componentName = request.params(PARAM_COMPONENT_NAME);
-        boolean isStrict = isStrictMode(request);
+        StatusCodeConvertMode convertMode = getStatusCodeConvertMode(request);
         HealthCheck healthCheck = healthChecks.stream()
             .filter(c -> c.componentName().getName().equals(componentName))
             .findFirst()
             .orElseThrow(() -> throw404(componentName));
 
         Result result = Mono.from(healthCheck.check()).block();
         logFailedCheck(result);
-        response.status(getCorrespondingStatusCode(result.getStatus(), isStrict));
+        response.status(getCorrespondingStatusCode(result.getStatus(), convertMode));
```

Comment:
Inline that: we definitly do not need that level of indirection

```suggestion
        response.status(convertMode.getCorrespondingSatus(result.getStatus())));
```